examples/query/react/basic/src/test/test-utils.tsx

```diff
@@ -0,0 +1,45 @@
+import { render as rtlRender } from '@testing-library/react'
+import type { RenderOptions } from '@testing-library/react'
+import React, { PropsWithChildren } from 'react'
+import { Provider } from 'react-redux'
+import { setupStore } from '../store'
+import type { AppStore, RootState } from '../store'
+import type { PreloadedState } from '@reduxjs/toolkit'
+
+// This type interface extends the default options for render from RTL, as well
+// as allows the user to specify other things such as initialState, store. For
+// future dependencies, such as wanting to test with react-router, you can extend
+// this interface to accept a path and route and use those in a <MemoryRouter />
+interface IWithStoreOptions extends Omit<RenderOptions, 'queries'> {
+  initialState?: PreloadedState<RootState>
+  store?: AppStore
+}
+
+function render(
+  ui: React.ReactElement,
+  {
+    initialState = {},
+    store = setupStore(initialState),
+    ...renderOptions
+  }: IWithStoreOptions = {}
```

Comment:
```suggestion
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialState?: PreloadedState<RootState>
  store?: AppStore
}

function render(
  ui: React.ReactElement,
  {
    initialState = {},
    store = setupStore(initialState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
```

I think this name (`ExtendedRenderOptions`) is a little clearer. The `I` prefix isn't really necessary with typescript, but that's not a big deal.

Omitting 'store' explicitly in the name makes it retain accuracy if it gets extended in the future with a bunch of other options unrelated to the store

Reply:
I agree, thanks!